src/cdk-lib/bedrock/agent.ts

```diff
@@ -80,6 +85,109 @@ export interface GuardrailConfiguration {
   readonly guardrailId?: string;
 }
 
+export interface IAgent extends IResource {
+
+  /**
+   * The name of the agent.
+   */
+  readonly name: string;
+
+  /**
+   * The IAM role for the agent.
+   */
+  readonly role: iam.Role;
```

Comment:
I think it would be better to have this as `IRole` rather than `Role`.

Reply:
Agree on this one

Reply:
Refactored